test/test_all.py

```diff
@@ -119,6 +119,21 @@ def check_libraries(configuration):
 
         assert release["checksum"] == "SHA-256:" + hashlib.sha256(release_archive_path.read_bytes()).hexdigest()
 
+    # Check against the db
+    with pathlib.Path(configuration["LibrariesDB"]).open(mode="r", encoding="utf-8") as library_db_file:
+        library_db = json.load(fp=library_db_file)
+    for release in library_db["Releases"]:
+        release_archive_path = pathlib.Path(
+            configuration["LibrariesFolder"],
+            release["URL"].removeprefix(configuration["BaseDownloadUrl"]),
+        )
+
+        assert release_archive_path.exists()
+
+        assert release["Size"] == release_archive_path.stat().st_size
```

Comment:
I don't know if we can compare the file size with a golder, the main concern is that `gzip`, as the algorithm evolves, may generate slightly different sizes in the output (even with the same input file).

IMHO we should either drop the check or allow an error within a delta (like +/-1%).

Reply:
Good point! Here it is:
https://github.com/arduino/libraries-repository-engine/compare/7ac77910a83f97e2e7aba20caa276978010b3eae..7b43c4e968e1da8c3fef928630df096fcae8228b
I decided I prefer to keep the check in because I think it can serve as a useful "smoke test" even if we aren't verifying the exact size is as expected. I set the tolerance to 3%. I don't know whether that is an appropriate value or not.

I definitely don't want this test to end up being something that breaks all the time. On the other hand, a size difference in the archives actually did alert me to a problem introduced by the switch to go-git where empty folders were being left behind in the library repository after checking out from a tag that used the folder to another that didn't use the folder. So I think it does have more value with a tighter tolerance.